### PR TITLE
fix: check exit code in avrogen required flags tests

### PR DESCRIPTION
**Pull request by hhromic**
While working on PR #302, I realized that the unit tests did _not_ fail when I removed the requirement for `-o FILE`.
The reason is that the unit tests were expecting errors but not _which_ errors. When I removed the requirement for `-o FILE`, the test still passed because `realMain` still failed but for a different reason (schema not found).

This PR fixes the avrogen required flags unit tests by checking the actual exit code returned from `realMain`.

### CODE REVIEW

File: cmd/avrogen/main_test.go [hunk 1]
```diff
@@ -16,39 +16,39 @@ var update = flag.Bool("update", false, "Update golden files")
 
 func TestAvroGen_RequiredFlags(t *testing.T) {
 	tests := []struct {
-		name    string
-		args    []string
-		wantErr bool
+		name     string
+		args     []string
+		exitCode int
```
Comment (nrwiersma) [inline cmd/avrogen/main_test.go:21]: Nit: it is typical to start test assertion properties with `want` to differentiate them from test input properties. I would suggest:
```suggestion
		wantExitCode int
```

**Review by nrwiersma — CHANGES_REQUESTED** [review]
A tiny nit in naming, but nice change over all.

**Review by nrwiersma — APPROVED** [review]
LGTM 🎉 Thanks.

File: cmd/avrogen/main_test.go (hunk 1 above)
Comment (hhromic) [reply cmd/avrogen/main_test.go]: Doh. Apologies, had a brain glitch with that one. 🤦 
Fixed now!

**Pull request merged**